### PR TITLE
Make zombies 2spooky

### PR DESCRIPTION
**Pull request by Robustin**
1) Double HP (now 200, so they actually take a little effort to mow down)
2) Can tear down airlocks in around 25 seconds, making super spooky sounds
3) They are now speed = 2, so you can outrun them. If they catch you, you kinda deserve it. (Rule #1 Cardio)

Zombies have potential but their biggest issue is how they immediately fuck up the round before anyone can really react. They function more like 28 weeks later infected (as fragile as humans but just as fast) than traditional zombies (slow, lumbering, takes a beating, but relentless). SS13 is better suited to the latter than a single zombie player that infects 80% of the station within 2 minutes, followed by 20 minutes where everyone else just hides behind welded doors until the round ends. 

This means zombies start slower but finish stronger. A couple zombies will be target practice, but if they manage to grow their numbers, they become infinitely spookier than a bunch of turbozombies zipping around but unable bypass a single door. 

This should make for overall more interesting rounds, I might reduce the revive timer if I think that this makes zombies too much of a non-threat.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 1]
```diff
@@ -51,6 +51,14 @@
 			visible_message("<span class='danger'>[src] tears [L] to pieces!</span>")
 			src << "<span class='userdanger'>You feast on [L], restoring your health!</span>"
 			revive(full_heal = 1)
+			
+	if(istype(target, /obj/machinery/door/airlock))
+		src << "<span class='notice'>You start tearing apart the airlock...</span>"
+		playsound(src.loc, 'sound/hallucinations/growl3.ogg', 50, 1)
+		if(do_after(src, 250, target))
+			playsound(src.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)
```
Comment (GunHog) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: Should be one of the loud banging noises, to properly indicate that it is time to add a little brown coloring to your jumpsuit.

**Comment by GunHog** [comment]
Did you test this by running the code locally, then playing one of these yourself?

**Comment by PKPenguin321** [comment]
Too much health imo

**Comment by GunHog** [comment]
In a melee fight, speed is king. You could be a 400 health tank with 50 damage, but that will mean little if you are so slow that you cannot catch up to anyone unless they are almost dead or trapped.

**Comment by RemieRichards** [comment]
I don't like the door being destroyed.
it should be down graded to just a construction frame of the door, so it looks like it's legitimately been torn open, rather than just *poof* no door

**Comment by Robustin** [comment]
I'll see if I can "code" (steal) the lines necessary to make the door breaking more subtle. 

As for the sound, I would need a decent example. I might consider flipping the noises so that the zombies wails when he starts on the door and groans when it goes down. Either way the wail is the clear pants-filled here, just thought it was more appropriate for a dramatic entrance.

**Comment by PKPenguin321** [comment]
Not that hard just spawn a new airlock frame at the airlock loc before you qdel it, it would be like a 1 line change

**Comment by Shadowlight213** [comment]
Aren't the frames dense though? Having the zombie tear the door down, then have to stop and pull/push the frame out of the way would kinda kill the mood.

**Comment by KazeEspada** [comment]
Reduce it to metal and wire then.

**Comment by Shadowlight213** [comment]
Nah, making the frame anchored and nondense would work fine if we got some visual distinction that it's torn down instead of not built.

**Comment by Robustin** [comment]
Spawning a special frame that's non-dense and anchored might be beyond my theft abilties.

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 2]
```diff
@@ -51,6 +51,14 @@
...
+			qdel(target)
```
Comment (PKPenguin321) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: before this line:
var/obj/structure/door_assembly/door = new(src.loc)
door.density = 0
door.anchored = 1

and this is optional:
door.name = "destroyed airlock"
door.desc = "A broken airlock. Looks like it wont be keeping much out now."

Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: Except that door assemblies are normally dense, and making them undense is a recipe for disaster.

Comment (PKPenguin321) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: I don't exactly see how

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: this, except ```new(src.loc)``` becomes ```new airlock.doortype(target.loc)```
1. Airlocks have specific assemblies, so they don't all look generic
2. src.loc would put it where the zombie is, that's weird

Comment (Robustin) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: I suppose all doors start to look alike once a zombie finishes tearing it apart.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: true, but a zombie wouldn't methodically remove every spec of paint on the door.

**Comment by Niknakflak** [comment]
Health is too high imo.  Zombies are weak falling apart corpses.  That's what makes them zombies.  :-1:

**Comment by Robustin** [comment]
This is a 2d spessman game. I made this change for balance and gameplay purposes, the analogies were just there to help people understand the atmosphere I was promoting.

Slowing them down means they should never get the consecutive hits necessary to zombify someone without cornering them with superior numbers. 200hp makes them hearty enough to not git rekt by the first person they see with a gun. 

Keep in mind reducing speed reduces their effective HP anyway. People will have a much easier time landing hits on a zombie that can't juke.

**Comment by PKPenguin321** [comment]
with 200 hp basically the only people that will ever fight the zombies are security/head roles who have access to firearms

at least with lower health you can still beat them with like a welder or floor tiles or something, giving them that much HP is just shit

how about keep HP low but make them regenerating much faster

**Comment by Robustin** [comment]
@PKPenguin321 

Or you know... anyone who can rip up floor tiles or grab fire extinguishers and throw them.

**Comment by PKPenguin321** [comment]
20 tiles is a pain in the dick though, especially if there is any more than 1 zombie
10 is more reasonable, you feel?

**Comment by Robustin** [comment]
Theyre slow now, and its 180hp now. 

Are people really forgetting how important speed is in this game? This is a zombie nerf, by and large, but at least they won't be the immediate game-enders they are now.

**Comment by Cheridan** [comment]
Zombies should be hard to kill, but instead of it being just 200 health to tank through, it should be more like... they have 70 health, but every time you knock it down it just gets back up and keeps coming, you hit it with a fireaxe and its arm falls off but the arm is still crawling at you.

**Comment by Robustin** [comment]
Well seeing as dismemberment is coming never, best alternative is just giving them a much shorter revive window, but if the crew's butchering the corpses it wont matter, anything under 100hp means that spearsistants will win 1v1 easy.

**Comment by Zilenan91** [comment]
Make it so glass shards can't butcher because that makes zombies way too easy to gib

**Comment by Robustin** [comment]
I see no reason why we can't merge this. HP/Revive stuff can be easily changed if they're too tanky (again its 180 now), but I really like the trade-off of doorstruction v. move speed.

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 3]
```diff
@@ -51,6 +51,19 @@
...
+			var/obj/structure/door_assembly/door = new(target.loc)
+			door.density = 0
+			door.anchored = 1
+			door.name = "ravaged airlock"
+			door.desc = "An airlock that has been torn apart. Looks like it wont be keeping much out now."
+			return
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: this return does nothing, please remove it.

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 4]
```diff
@@ -51,6 +51,19 @@
...
 			src << "<span class='userdanger'>You feast on [L], restoring your health!</span>"
 			revive(full_heal = 1)
+			
+	if(istype(target, /obj/machinery/door/airlock))
+		src << "<span class='notice'>You start tearing apart the airlock...</span>"
+		playsound(src.loc, 'sound/hallucinations/growl3.ogg', 50, 1)
+		if(do_after(src, 250, target))
+			playsound(src.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)
+			qdel(target)
+			var/obj/structure/door_assembly/door = new(target.loc)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: swap this line for:
```
var/obj/machinery/door/airlock/A = target
var/obj/structure/door_assembly/door = new A.doortype(target.loc)
```

**Comment by RemieRichards** [comment]
@Robustin please fix what I've pointed out, and numbers/balance memes or whatever else is holding this up can continue.

**Comment by Robustin** [comment]
I made those fixes, and I'll reiterate that I'd really like to see this merged so I can update my fork and move ahead with several other changes I have planned.

**Comment by RemieRichards** [comment]
@Robustin so long as your work on your branch is fully committed locally (which to have pushed here, it really should be) you can swap branches to master, update your master to the latest, and then make new branches to keep working, you DON'T have to wait on this PR.

**Comment by Robustin** [comment]
Thanks. To the chagrin of others, I pretty much use the Web stuff exclusively. When I did the big cult PR I tried to do it "by the books" but got bogged down in a dozen little issues that never did get completely solved even after consulting virtually everyone. I'm convinced that just keeping my PR's relatively small, simple, and online (I do most of this stuff when I have downtime at work) is a good fit for me.

**Comment by GunHog** [comment]
Robustin, you need to compile and test your code, do not just throw up a PR and make Travis and the maintainers do your work for you. If you absolutely cannot get Git to work for you on your computer, then copy over the changes you make in DM by erasing the online version of the file and pasting the entire modified version from DM. I did this for a while before I took the time to wrangle the desktop version.

**Pull request merged**